python/tvm/relay/ty.py

```diff
@@ -106,6 +122,51 @@ def __init__(self, var, kind=Kind.Type):
         self.__init_handle_by_constructor__(_make.TypeVar, var, kind)
 
 
+@register_relay_node
+class GlobalTypeVar(Type):
+    """A global type variable in Relay.
+    GlobalTypeVar is used to refer to the global type-level definitions
+    stored in the environment.
+    """
+
+    def __init__(self, var, kind=Kind.AdtHandle):
+        """Construct a GlobalTypeVar.
+
+        Parameters
+        ----------
+        var: tvm.Var
+            The tvm.Var which backs the type parameter.
+        kind: Kind, optional
+            The kind of the type parameter, Kind.AdtHandle by default.
+
+        Returns
+        -------
+        type_var: GlobalTypeVar
+            The global type variable.
+        """
+        self.__init_handle_by_constructor__(_make.GlobalTypeVar, var, kind)
+
+
+@register_relay_node
+class TypeCall(Type):
+    """Type-level function application in Relay."""
```

Comment:
I think we should change the docstring to reflect that this is the application of a type constructor to type arguments. Or at least make that clear to people.

Reply:
Good idea